src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9091,6 +9093,10 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                 {
                     opKind = SyntaxKind.SwitchExpression;
                 }
+                else if (tk == SyntaxKind.WithKeyword && this.PeekToken(1).Kind == SyntaxKind.OpenBraceToken)
```

Comment:
do you even need the open brace?  it couldn't be legal otherwise, right?  so maybe just try to parse here and say that the `{` is missing.

Reply:
Maybe? I don't know. I can't think of how it would be ambiguous, but maybe I'm just not thinking of something :)